_posts/2018-03-26-proxstar.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: post
+title: Proxstar
+date: 2018-03-26
+categories: projects
+description: Proxstar is a VM self-service tool that integrates with Proxmox and STARRS
+author: Jordan Rodgers
+author-image: https://avatars1.githubusercontent.com/u/5183608?s=460&v=4
+author-bio: Fourth Year Networking & Systems Administration Student at Rochester Institute of Technology
+author-email: com6056@csh.rit.edu
+author-social:
+  github: https://github.com/com6056
+---
+
+## The Problem
+CSH currently uses Proxmox for VM hosting. It has been working great for everything we use it work with one exception: member VM self-service. The interface isn't very intuitive to someone who might not have created a VM before, creating a higher barrier to entry than we would like for members who might want to start learning how to use Linux for the first time. It also lacked the ability to enforce resource limits and automatically cleanup VMs that members weren't using anymore. All of my previous major projects were much more ops and/or security focused so I figured writing something to solve this problem would be a great way to do a much more development focused one and expand my developer skillset.
+
+## The Solution
+Proxstar (a mashup of Proxmox and STARRS) is a web application that allows for easy VM self-service for the members of CSH. It is written in Python using Flask and runs on the CSH OpenShift cluster. RQ is being used as a task queue for tasks that take more than a second or two and periodic tasks (RQ-Scheduler). Proxstar allowed members to create, edit, and delete VMs and will automatically handle any network registration that needs to be done in STARRS for them. During creation, they can specify to either create it from a template or to install it themselves from an ISO. If creating it from a template, Proxstar will spin up a ready-to-go VM with a a user that has a randomly generated password for the member to use.
```

Comment:
Maybe "RQ is being used as a task queue for schedule-tasks (via RQ-Scheduler) and long-running tasks." or similar.

Reply:
`s/allowed/allows/`

Reply:
`s/specify to//`
`s/create it/create the VM/`

Reply:
`s/creating it/creating a VM/`
`s/a a user/a user/`